### DISCUSSION TITLE
Facing issue while using srsEnodeB and srsUE

### DISCUSSION THREAD

**Discussion by shalinjshah18** [magma/magma#7909] [Q&A]
@hesseltjeerdsma  Hi, 

We have setup of Magma AG on Virtual Machine;
Configured srsUE and srsEnodeB to point to Magma AGW on Virtual Machine.

But seeing error when we run them up
srsenb    | bind(): Cannot assign requested address
srsenb    | connect(): Connection timed out
srsenb    | Failed to initiate S1 connection. Attempting reconnection in 10 seconds
srsenb    | Failed to bind on address 192.168.128.1, port 2152: Cannot assign requested address
srsenb    | Error initializing stack.

AGW eth0 10.100.52.XXX/24, eth1 10.0.2.1/24 running on Ubuntu 20.04
gtp_br0: 192.168.128.1/24

srsUE and srsEnodeB setup on Ubuntu 20.04 VM
ens160: 10.100.52.XXX/24

Configuration Inside enb.conf file is:

mme_addr = 10.0.2.1
# gtp s1c are gateway IP for the eNodeB interface
gtp_bind_addr = 10.100.52.XXX (This is VM IP of srsUE and srsEnodeB)
s1c_bind_addr = 10.100.52.XXX (This is VM IP of srsUE and srsEnodeB)

Once we run srsEnodeB using ./srsenb command it throws and error unknown PLMN.
Need guidance how to further configure Magma NMS UI, eNodeB config in NMS.

**Comment by HesselTjeerdsma** [comment]
Did you change the MCC en MNC in the magmacore to match the ones specified in SRS config? The PLMN error seems to indicate that SRSenb is able to talk to the AGW, but the connection is rejected due to MCC en MNC not matching. 

Do you also see the bind error when connecting to the AGW?

**Reply by shalinjshah18** [reply]
Yes we changed MCC and MNC in MagmaCore and Match it with SRS config.
We saw bind error earlier but than we change gtp_br IP to local VM IP and than it went away.
Earlier we were using 192.168.128.1 as gtp_br IP in srs config.

**Reply by HesselTjeerdsma** [reply]
Did you check the TAC?

**Reply by shalinjshah18** [reply]
We have configured TAC value in Magma NMS but how do we configure that value in srsRAN/srsEnodeb/srsUE?

**Reply by Radula76** [reply]
srsEnb rr.conf file:
cell_list =
(
  {
    // rf_port = 0;
    cell_id = 0x01;        cell id
    tac = 0x0007;         TAC code, 7 in my case which is what is in my AGW Network from NMS/Orc8r.

**Comment by Radula76** [comment]
you need to run the srsue process as root, it needs to create the virtual tunnel endpoint/bridge.

**Reply by shalinjshah18** [reply]
[PipelineD Errors.txt](https://github.com/magma/magma/files/6776104/PipelineD.Errors.txt)
 @Radula76 and @HesselTjeerdsma 
We setup srsUE and eNode B request is reaching to Magma AGW as well; But than inside AGW we see errors related to PipelineD Service. Attached are logs for the same.
Please help if this is some misconfiguration in Magma AGW side/its related to srsUE/srsEnodeB config issue.

**Reply by pshelar** [reply]
@shalinjshah18 what is status of ovs services ? `sudo service openvswitch-switch status` and `sudo ovs-vsctl show`

**Reply by shalinjshah18** [reply]
[OpenVSwitch Service Status.txt](https://github.com/magma/magma/files/6778789/OpenVSwitch.Service.Status.txt)
![Network Status after Magma Services Restarted](https://user-images.githubusercontent.com/83823581/124797067-7d363300-df6f-11eb-8fa1-4a825b97ee65.png)
![Network Status before Magma Services Restart](https://user-images.githubusercontent.com/83823581/124797073-7f988d00-df6f-11eb-873a-5fd6dcdc5b59.png)

Please find attached file which contains status of sudo service openvswitch-switch status and sudo ovs-vsctl show
Also please find two images Network Status before Magma Services Restart.png is image that shows network interfaces as normal

eth0 - 10.100.52.XXX/24 
eth1 - 10.0.2.1/24 

But after we execute sudo service magma@magmad restart - network eth0 and eth1 gets altered - refer to Network Status after Magma Services Restarted.png and both ethernet cards will have 10.0.2.1/ IP with /32 and /24 subnet respectively.

This is surprising since AGW eth0 is SGI interface towards internet and it get altered after AGW restart.

**Reply by pshelar** [reply]
since there is not static IP defined via API, AGW is starting dhclient for eth0.

**Reply by pshelar** [reply]
The `ovs-vsctl show` is flagging error: `error: "could not add network device gtp0 to ofproto (Address family not supported by protocol)"`
It means GTP support is not loaded, can you run `ovs-kmod-upgrade.sh` and post output of the script.

**Reply by shalinjshah18** [reply]
> The `ovs-vsctl show` is flagging error: `error: "could not add network device gtp0 to ofproto (Address family not supported by protocol)"`
> It means GTP support is not loaded, can you run `ovs-kmod-upgrade.sh` and post output of the script.

Hi Pravin, We went ahead and ran this script from /home/admin/magma/lte/gateway/deploy/roles/magma/files folder. Seems it failed and has errors.
[Output of ovs-kmod-upgrade.txt](https://github.com/magma/magma/files/6781892/Output.of.ovs-kmod-upgrade.txt)

**Reply by shalinjshah18** [reply]
> since there is not static IP defined via API, AGW is starting dhclient for eth0.

We earlier had configuration in NMS UI inside Gateway -> EPC tab from Edit Gateway section.
SGi network Gateway IP address: 10.100.52.XXX
SGi management interface IP address: 10.0.2.1 
We removed both of them from UI and than restarted AGW.  It seems now IPs are stable. But will repost here in case they again resets.

**Reply by shalinjshah18** [reply]
Hi Pravin,
Please find attached output of  cat /var/lib/dkms/openvswitch/2.14.3/build/make.log

[make.log](https://github.com/magma/magma/files/6781929/make.log)
it shows error 

configure: error: Linux kernel in /lib/modules/5.8.0-59-generic/build is version 5.8.18, but version newer than 5.6.x is not supported (please refer to the FAQ for advice)

**Reply by shalinjshah18** [reply]
Hi Pravin,

We were able to run ovs-kmod-upgrade.sh successfully.
After this, we restarted AGW and tried using srseNodeB and srsUE to attach to AGW.
We are getting successful attach as well. But we observe below error and in couple of seconds UE gets disconnected.

AGW Error:

Jul  8 16:33:27 Ubuntu-20 sessiond[35615]: I0708 16:33:27.672387 35615 LocalEnforcer.cpp:1990] Failed mac/name parsiong for apn internet

srsue     | RRC Connected
srsue     | Random Access Complete.     c-rnti=0x46, ta=0
srsue     | Network attach successful. IP: 192.168.128.17
srsenb    | User 0x46 connected
srsue     | E
srsue     | Received RRC Connection Release (releaseCause: other)
srsue     | RRC IDLE
srsenb    | Disconnecting rnti=0x46.
srsue     | S-TMSI match in paging message
srsue     | Service Request with cause mt-Access.
srsue     | Random Access Transmission: seq=15, tti=1191, ra-rnti=0x2
srsenb    | RACH:  tti=1191, cc=0, preamble=15, offset=0, temp_crnti=0x47
srsue     | RRC Connected
srsue     | Random Access Complete.     c-rnti=0x47, ta=0
srsue     | Service Request successful.
srsenb    | User 0x47 connected
srsue     | Received RRC Connection Release (releaseCause: other)
srsue     | RRC IDLE
srsenb    | Disconnecting rnti=0x47.

This data upload/download is not happening using iperf3 -s and than iperf3 -c 192.168.128.17

**Reply by Radula76** [reply]
those look normal.  the eNB will drop the radio links when there is no traffic:
RACH:  tti=1361, cc=0, preamble=3, offset=0, temp_crnti=0x49
User 0x49 connected
Disconnecting rnti=0x49.

same with the Ue going idle:
Random Access Transmission: seq=3, ra-rnti=0x2
Random Access Complete.     c-rnti=0x49, ta=0
RRC Connected
Received RRC Connection Release (releaseCause: other)
RRC IDLE

you should still be able to send traffic in this state.
your iperf3 -s is running on what IP address?  the iperf3 -c should be listing the server address, not your own IP address.
